scripts/core.js

```diff
@@ -17,36 +17,38 @@
 /**
  *  Represents a single substitution with an input, an output and some additional parameters.
  */
-var Substitution = (() => {
+const Substitution = (() => {
 
   class Substitution {
 
-    constructor(input, output, caseSensitive = false, inputType = this.INPUT_TEXT) {
+    constructor(input, output, caseSensitive = false, inputType = this.INPUT_TEXT, outputType = this.OUTPUT_FUNCTION) {
       this.input = input;
       this.output = output;
-      this.caseSensitive = Boolean(caseSensitive);
-      this.inputType = +inputType;  // coalesce to number
+      this.caseSensitive = !!caseSensitive;
+      this.outputType = +outputType;
+      this.inputType = +inputType;
+
       if (this.inputType < this.INPUT_TEXT || this.inputType > this.INPUT_REG_EXP) this.inputType = this.INPUT_TEXT;  // avoid invalid values
```

Comment:
Maybe you could add another line like this for the output type, to ensure it is a valid value.